roles/StackStorm.st2smoketests/tasks/st2rbac_verification.yml

```diff
@@ -0,0 +1,142 @@
+---
+
+# Small suite of smoke tests to ensure that RBAC works as expected
+- name: Create directory to store roles and assignments
+  become: yes
+  file:
+    path: "/opt/stackstorm/rbac/{{ item }}"
+    recurse: yes
+    state: directory
+  loop:
+    - roles
+    - assignments
+
+- name: auth | Add a new smoke_tests_rbac_user in st2 htpasswd file
+  become: true
+  htpasswd:
+    path: /etc/st2/htpasswd
+    name: "{{ smoke_tests_rbac_user }}"
+    password: "{{ smoke_tests_rbac_password }}"
+  changed_when: no
+  notify:
+    - restart st2api
+    - restart st2stream
+
+- name: Copy smoke tests RBAC roles to /opt/stackstorm/rbac/roles directory
+  become: yes
+  template:
+    src: rbac_roles.yml.j2
+    dest: /opt/stackstorm/rbac/roles/{{ item.name }}.yaml
+    owner: st2
+    group: st2
+  loop: "{{ smoke_tests_rbac.roles }}"
+  when: smoke_tests_rbac.roles is defined
+  notify:
+    - reload rbac
+
+- name: Copy smoke tests RBAC assignments to /opt/stackstorm/rbac/assignments directory
+  become: yes
+  template:
+    src: rbac_assignments.yml.j2
+    dest: /opt/stackstorm/rbac/assignments/{{ item.name }}.yaml
+    owner: st2
+    group: st2
+  loop: "{{ smoke_tests_rbac.assignments }}"
+  when: smoke_tests_rbac.assignments is defined
+  notify:
+    - reload rbac
+
+- name: Enable RBAC in st2 configuration
+  become: yes
+  ini_file:
+    dest: /etc/st2/st2.conf
+    section: rbac
+    option: enable
+    value: True
+    backup: yes
+  notify:
+    - restart st2api
+    - restart st2auth
+    - reload rbac
```

Comment:
I guess the smoke test should really check if RBAC was configured properly and works if we set the required values in the role via vars + some specific settings for the test.

Creating directories, modifying st2.conf should remain as part of the original st2 role itself. Otherwise, we're not testing the original role behavior.

With that, I think you want to set `st2_rbac_enable: yes` that would run in CI environment only.
Similar to what was done here: https://github.com/StackStorm/ansible-st2/pull/290/files#diff-305ef63bd067b0c2e4aaae3beda8ae5ecb8d0bc9e490de22568e3e258184e50dL19-L20
and let the rbac part of the role to configure everything.

Reply:
Ok, that's what I had in the first place but then I thought it's less invasive to test RBAC just as part of the smoketests. But your point is totally valid, in the first place we want to test the setup not the RBAC functionality of st2. 

I moved the code to a dedicated local branch before I switched to the "wrong" solution. So I'll switch back to the original idea. 

I'll mark the PR as WIP until we pass the tests and it's ready for review.